docs/CMakeLists.txt

```diff
@@ -50,20 +48,17 @@ set(DOXYGEN_TOC_INCLUDE_HEADINGS 4)
 set(DOXYGEN_TYPEDEF_HIDES_STRUCT YES)
 set(DOXYGEN_UML_LIMIT_NUM_FIELDS 20)
 set(DOXYGEN_UML_LOOK YES)
-set(DOXYGEN_WARN_NO_PARAMDOC YES)
-set(DOXYGEN_WARN_IF_UNDOCUMENTED YES)
+set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
 set(DOXYGEN_WARN_IF_DOC_ERROR YES)
-set(DOXYGEN_WARN_AS_ERROR NO)
+set(DOXYGEN_WARN_IF_INCOMPLETE_DOC YES)
+set(DOXYGEN_WARN_NO_PARAMDOC YES)
+set(DOXYGEN_WARN_IF_UNDOC_ENUM_VAL YES)
+set(DOXYGEN_WARN_AS_ERROR FAIL_ON_WARNINGS)
 set(DOXYGEN_WARN_LOGFILE "doxygen.log")
 set(DOXYGEN_IMAGE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/images")
-
-FILE(GLOB_RECURSE HPP_SOURCES
-    "${DDC_SOURCE_DIR}/include/ddc/*.hpp")
-message(STATUS ${HPP_SOURCES})
```

Comment:
This was added to allow me to do `set(DOXYGEN_EXTRACT_ALL NO)`. Without this doxygen does not raise warnings